src/openms/include/OpenMS/CHEMISTRY/TheoreticalSpectrumGeneratorXLMS.h

```diff
@@ -166,14 +193,68 @@ namespace OpenMS
+       * @param charge The charge of the precursor
+       */
+      virtual void addPrecursorPeaks_(PeakSpectrum & spectrum, DataArrays::IntegerDataArray & charges, DataArrays::StringDataArray & ion_names, double precursor_mass, int charge) const;
+
+      /**
+       * @brief Adds losses for a linear ion
+
+       * @param spectrum The spectrum to which the new peak is added
+       * @param charges A DataArray collecting the charges of the added peaks
+       * @param ion_names A DataArray collecting the ion names of the added peaks
+       * @param ion
+       * @param res_type The ion type of the added peak
+       * @param frag_index The index of the ion (fragmentation position)
+       * @param intensity
+       * @param charge The charge of the ion
+       * @param ion_type Another cross-linking specific ion-type
+       */
+      virtual void addLinearIonLosses_(PeakSpectrum & spectrum, DataArrays::IntegerDataArray& charges, DataArrays::StringDataArray& ion_names, const AASequence & ion, Residue::ResidueType res_type, Size frag_index, double intensity, int charge, String ion_type) const;
+
+      /**
+       * @brief Adds one-residue-linked ion peaks, that are specific to XLMS
+
+          These fragments consist of one whole peptide, the cross-linker and a part of the linked residue from the second peptide.
+          The residue fragment on the linker is an internal ion from a y- and an a-fragmentation with the length of one residue.
+          The function is called KLinked for now, but instead of K it is whatever the linker is attached to.
+
+       * @param spectrum The spectrum to which the peaks are added
+       * @param charges A DataArray collecting the charges of the added peaks
+       * @param ion_names A DataArray collecting the ion names of the added peaks
+       * @param peptide The fragmented peptide
+       * @param link_pos position of the linker on the fragmented peptide
+       * @param precursor_mass The mass of the whole cross-link candidate or the precursor mass of the experimental MS2 spectrum.
+       * @param frag_alpha True, if the fragmented peptide is the Alpha peptide. Used for ion-name annotation.
+       * @param charge The charge of the ion
+       */
+      virtual void addKLinkedIonPeaks_(PeakSpectrum & spectrum, DataArrays::IntegerDataArray & charges, DataArrays::StringDataArray & ion_names, AASequence & peptide, Size link_pos, double precursor_mass, bool frag_alpha, int charge) const;
+
+
+      virtual void addComplexXLinkIonPeaks_(PeakSpectrum & spectrum, DataArrays::IntegerDataArray & charges, DataArrays::StringDataArray & ion_names, OPXLDataStructs::ProteinProteinCrossLink & crosslink, int charge, std::vector< std::set< LossMass > > & forward_losses_alpha, std::vector< std::set< LossMass > > & backward_losses_alpha, std::vector< std::set< LossMass > > & forward_losses_beta, std::vector< std::set< LossMass > > & backward_losses_beta) const;
```

Comment:
add missing documentation (here and else but I guess this will come with some additional commits)